modules/core/src/lib/base-attribute.js

```diff
@@ -60,7 +58,17 @@ export default class BaseAttribute {
       }
     } else if (value) {
       this.externalBuffer = null;
-      this.value = value;
+
+      const size = this.size || opts.size || 0;
+      if (constant && value.length !== size) {
+        this.value = new Float32Array(size);
+        const index = this.offset / 4;
```

Comment:
It can only every be 4 bytes (float, int or uint). We currently only use floats, but I'll put a note about a gotcha here about the fact that if we ever start using int attributes (WebGL 2-only), this will need to be fixed.